lib/ostruct.rb

```diff
@@ -88,12 +87,13 @@ class OpenStruct
   #
   #   data   # => #<OpenStruct country="Australia", capital="Canberra">
   #
-  def initialize(hash=nil)
+  def initialize(hash=nil, options={recursive: false})
     @table = {}
+    @recursive = options.fetch(:recursive, false)
     if hash
       hash.each_pair do |k, v|
         k = k.to_sym
-        @table[k] = v
+        @table[k] = (@recursive && v.respond_to?(:to_hash)) ? OpenStruct.new(v,true) : v
```

Comment:
`OpenStruct.new(v, options) ` or `OpenStruct.new(v,  recursive: true) `

Reply:
Yes, it should be `OpenStruct.new(v, options)`. Will update the PR in a bit.